core/src/main/java/org/apache/brooklyn/core/catalog/internal/BasicBrooklynCatalog.java

```diff
@@ -800,7 +809,11 @@ private boolean attemptType(String key, CatalogItemType candidateCiType) {
                     // (when we're not given a key, the previous block should apply)
                     errors.add(e);
                 } else {
-                    // all other cases, the error is probably due to us not getting the type right, ignore it
+                    // all other cases, the error is probably due to us not getting the type right, probably ignore it
+                    // but cache it if we've checked entity, we'll use that as fallback errors
```

Comment:
Not sure I follow - why would we get the type wrong for non-entities? Have we guessed that it was an entity, without being able to tell from the context if its instead a policy/enricher/etc?